File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemPagerCardVModel.java
package io.ganguo.demo.viewmodel.item;

import android.databinding.ObservableField;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ItemPagerCardBinding;
import io.ganguo.demo.entity.ImageEntity;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.library.ui.view.ViewInterface;

/**
 * Created by leo on 16/7/31.
 * Item Pager-card
 */
public class ItemPagerCardVModel extends BaseViewModel<ViewInterface<ItemPagerCardBinding>> {
    private ImageEntity image;
    private ObservableField<String> name = new ObservableField<>();
    private ObservableField<String> imageUrl = new ObservableField<>();

    public ItemPagerCardVModel(ImageEntity image) {
        this.image = image;
        this.name.set(image.getName());
        this.imageUrl.set(image.getImage());
    }

    @Override
    public void onViewAttached(View view) {

    }

    public ObservableField<String> getName() {
        return name;
    }

    public ObservableField<String> getImageUrl() {
        return imageUrl;
    }

    @Override
    public int getItemLayoutId() {
        return R.layout.item_pager_card;
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemPagerCardVModel.kt
package io.ganguo.demo.viewmodel.item

import androidx.databinding.ObservableField
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemPagerCardBinding
import io.ganguo.demo.entity.ImageEntity
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.ViewInterface

/**
 * Created by leo on 16/7/31.
 * Item Pager-card
 */
class ItemPagerCardVModel(private val image: ImageEntity) : BaseViewModel<ViewInterface<ItemPagerCardBinding>>() {
    val name = ObservableField<String>()
    val imageUrl = ObservableField<String>()

    init {
        this.name.set(image.name)
        this.imageUrl.set(image.image)
    }

    override fun onViewAttached(view: View) {

    }

    override val layoutId: Int by lazy {
        R.layout.item_pager_card
    }
}
